compose/service.py

```diff
@@ -519,6 +522,59 @@ def pull(self, insecure_registry=False):
             )
 
 
+def get_container_data_volumes(container, volumes_option):
+    """Find the container data volumes that are in `volumes_option`, and return
+    a mapping of volume bindings for those volumes.
+    """
+    volumes = []
+
+    volumes_option = volumes_option or []
+    container_volumes = container.get('Volumes') or {}
+    original_volumes = set(volumes_option + container_volumes.keys())
+
+    for volume in original_volumes:
```

Comment:
@cpuguy83 Per @dnephin request adding info from my case here: https://gist.github.com/andreyst/e213c18ef32dcb28dc5e

This works if I try to mount something else like /tmp. If I ln -s the dir I need in /tmp and try to pass /tmp/dirname as a volume, it still doesn't work. If then I pass /tmp as a volume, it does work, but if then I try to cd to it from a container, it gives me "No such file or directory".